### PR TITLE
feat: [WIP] Integrate Matomo analytics tracking

### PR DESCRIPTION
**Pull request by philippeauriach**
## PR Description

Type of Changes 
- [X] New feature

⚠️ Nothing is operationel yet, as we don't have the Matomo server for now

Proposed changes
  - integrates anonymous Matomo analytics tracking
  - ask the user at first open if he agreed to be tracked (default to false)
  - privacy screen in setting allowing to enable / disable tracking
  - in the new priivacy screen, allow disabling crash reporting
 
views tracked :
- scanner view
products/edit
- events tracked :
  - scanner scanned (with barcode as event name)
  - scanner result card swiped up (with barcode as event name)
- allergen alert added (with allergen code as event name)
- ingredient-analysis
  - enabled (with name of ingredient analysis)
  - disabled (with name of ingredient analysis)
- user-account
  - login
  - logout
  - login prompt
    - name: robotoff
  - logged-in-after-prompt
    - name: robotoff
- products :
  - edited (saved a product) (with barcode as event name)
  - edited-nutrition-facts (saved a product’s nutrition facts table) (with barcode as event name)
  - edited-ingredients (saved a product’s ingredients) (with barcode as event name)
  - edited-ingredients-picture (added a product’s ingredients picture) (with barcode as event name)

### After
Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-19 at 18 12 34
Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-19 at 18 12 50

### CODE REVIEW

**Review by stephanegigandet — COMMENTED** [review]
Looks great :)

**Comment by philippeauriach** [comment]
We still miss the matomo server url. @teolemon what's the take on this?

**Comment by teolemon** [comment]
@cquest do you have some info for @philippeauriach regarding the matomo path for the app ? Any ETA ?

**Comment by teolemon** [comment]
@philippeauriach I reckon clicking that button is safe ?
image

**Comment by philippeauriach** [comment]
> @philippeauriach I reckon clicking that button is safe ?
> 
> image
> 
> 

I like this button! Should be yes, a conflictless merge theorically works anytime.

**Comment by teolemon** [comment]
Like the Cavalry, we arrive late, but we now have a Matomo install : https://analytics.openfoodfacts.org/

**Comment by teolemon** [comment]
Update branch attempt failed
Merge conflict between head and base.

**Comment by teolemon** [comment]
https://github.com/openfoodfacts/openfoodfacts-infrastructure/issues/24

File: Sources/Controllers/AnalyticsManager.swift [hunk 1]
```diff
@@ -5,23 +5,120 @@
...
 
 class AnalyticsManager {
 
-    static func log(_ value: String, forKey key: String) {
-        SentrySDK.configureScope { (scope: Scope) in
-            scope.setValue(value, forKey: key)
+    //TODO: config
+    let sharedMatomoTracker: MatomoTracker = MatomoTracker(
+        siteId: "1",
+        baseURL: URL(string: "https://example.com/piwik.php")!
```
Comment (teolemon) [inline Sources/Controllers/AnalyticsManager.swift]:
```suggestion
        baseURL: URL(string: "https://analytics.openfoodfacts.org/")!
```

Comment (teolemon) [reply Sources/Controllers/AnalyticsManager.swift]: @philippeauriach we have a working instance

**Comment by teolemon** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/openfoodfacts/openfoodfacts-ios) found multiple issues last seen at f844be12db45fb3dd0a04904fc9e6f075d13901b:

#### Lines should not have trailing whitespace.
- [ ]  [Sources\ViewControllers\Products\Add\ProductAddViewController.swift:359](https://github.com/openfoodfacts/openfoodfacts-ios/blob/f844be12db45fb3dd0a04904fc9e6f075d13901b/Sources/ViewControllers/Products/Add/ProductAddViewController.swift#L359)

#### Complex Method
- [ ]  [Sources\ViewControllers\Products\Detail\ProductDetailViewController.swift:248](https://github.com/openfoodfacts/openfoodfacts-ios/blob/f844be12db45fb3dd0a04904fc9e6f075d13901b/Sources/ViewControllers/Products/Detail/ProductDetailViewController.swift#L248)

#### TODOs should be resolved (config).
- [ ]  [Sources\Controllers\AnalyticsManager.swift:16](https://github.com/openfoodfacts/openfoodfacts-ios/blob/f844be12db45fb3dd0a04904fc9e6f075d13901b/Sources/Controllers/AnalyticsManager.swift#L16)

#### Prefer at least one space after slashes for comments.
- [ ]  [Sources\Controllers\AnalyticsManager.swift:16](https://github.com/openfoodfacts/openfoodfacts-ios/blob/f844be12db45fb3dd0a04904fc9e6f075d13901b/Sources/Controllers/AnalyticsManager.swift#L16)

**Review by teolemon — APPROVED** [review]

**Comment by teolemon** [comment]
Thank you @philippeauriach 
Given the merge conflict, and the fact that we moved to flutter, I'm closing this PR.

**Pull request closed without merging**